src/dag_algo/mod.rs

```diff
@@ -343,41 +343,47 @@ pub fn layers(
-/// node :math:`u` to node :math:`v`, :math:`u` comes before :math:`v`
-/// in the ordering.  If ``reverse`` is set to ``False``, the edges are treated
-/// as if they pointed in the opposite direction.
-///
-/// This function differs from :func:`~rustworkx.topological_sort` because
-/// when there are ties between nodes in the sort order this function will
-/// use the string returned by the ``key`` argument to determine the output
-/// order used.  The ``reverse`` argument does not affect the ordering of keys
-/// from this function, only the edges of the graph.
-///
-/// :param PyDiGraph dag: The DAG to get the topological sorted nodes from
-/// :param callable key: key is a python function or other callable that
-///     gets passed a single argument the node data from the graph and is
-///     expected to return a string which will be used for resolving ties
-///     in the sorting order.
-/// :param bool reverse: If ``False`` (the default), perform a regular
-///     topological ordering.  If ``True``, return the lexicographical
-///     topological order that would have been found if all the edges in the
-///     graph were reversed.  This does not affect the comparisons from the
-///     ``key``.
-/// :param Iterable[int] initial: If given, the initial node indices to start the topological
-///     ordering from.  If not given, the topological ordering will certainly contain every node in
-///     the graph.  If given, only the ``initial`` nodes and nodes that are dominated by the
-///     ``initial`` set will be in the ordering.  Notably, any node that has a natural in degree of
-///     zero will not be in the output ordering if ``initial`` is given and the zero-in-degree node
-///     is not in it.  It is a :exc:`ValueError` to give an `initial` set where the nodes have even
-///     a partial topological order between themselves.
-///
-/// :returns: A list of node's data lexicographically topologically sorted.
-/// :rtype: list
+
+/// Get the lexicographical topological sorted nodes from the provided directed
+/// graph.
+///
+/// This function returns a list of node data from the graph, sorted in
+/// lexicographical order based on the provided key function. A topological sort
+/// is a linear ordering of vertices such that for every directed edge from node
+/// :math:`u` to node :math:`v`, :math:`u` appears before :math:`v` in the
+/// ordering.  If `reverse` is set to `False`, the edges are treated as if they
+/// point in the opposite direction.
+///
+/// Unlike :func:`~rustworkx.topological_sort`, this function resolves ties
+/// between nodes using the string returned by the `key` argument. The `reverse`
+/// argument only affects the direction of the edges, not the ordering of keys.
+///
+///   >>> G = rx.PyDiGraph()
+///   >>> G.add_nodes_from(["A", "B", "C", "D", "E", "F", "G"])
+///   >>> G.add_edges_from_no_data([(0, 6), (1, 6), (2, 6), (3, 6), (4, 6), (5, 6)])
```

Comment:
Nitpicking, but you can unpack this as:
```
a, b, c, d, e, f, g = .add_nodes_from(["A", "B", "C", "D", "E", "F", "G"])
```

And then do e.g. `edges_from_no_data[(a, g)...]`

Reply:
In this case it seems to be useful. Updated, thanks.